java/ql/src/semmle/code/java/security/XPath.qll

```diff
@@ -0,0 +1,64 @@
+/** Provides classes to reason about XPath vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+
+/** The interface `javax.xml.xpath.XPath` */
+private class XPath extends Interface {
+  XPath() { this.hasQualifiedName("javax.xml.xpath", "XPath") }
+}
+
+/** A call to methods of any class implementing the interface `XPath` that evaluate XPath expressions */
+private class XPathEvaluation extends MethodAccess {
+  XPathEvaluation() {
+    exists(Method m |
+      this.getMethod() = m and m.getDeclaringType().getASourceSupertype*() instanceof XPath
+    |
+      m.hasName(["evaluate", "evaluateExpression", "compile"])
+    )
+  }
+
+  Expr getSink() { result = this.getArgument(0) }
+}
+
+/** The interface `org.dom4j.Node` */
+private class Dom4JNode extends Interface {
+  Dom4JNode() { this.hasQualifiedName("org.dom4j", "Node") }
+}
+
+/** A call to methods of any class implementing the interface `Node` that evaluate XPath expressions */
```

Comment:
```suggestion
/** A call to a method of any class implementing the interface `Node` that evaluate `XPath` expressions. */
```

Reply:
In aa10a5d, I refactored the library to use the CSV sink format, so this comment was removed.